File: vben-admin/apps/web-antd/src/api/ext-menu.ts
import { requestClient } from '#/api/request';

export interface ExtMenuItem {
  id: number;
  title: string;
  icon: string;
  url: string;
  sort_order: number;
  visible: number;
  scope: string;
  created_at: string;
}

/** 获取可见的扩展菜单（公开接口） */
export function getExtMenusPublicApi(): Promise<ExtMenuItem[]> {
  return requestClient.get('/ext-menus');
}

/** 管理员：获取所有扩展菜单 */
export function getExtMenusApi(): Promise<ExtMenuItem[]> {
  return requestClient.get('/admin/ext-menus');
}

/** 管理员：保存扩展菜单 */
export function saveExtMenuApi(data: Partial<ExtMenuItem>) {
  return requestClient.post('/admin/ext-menu/save', data);
}

/** 管理员：批量更新扩展菜单排序 */
export function reorderExtMenusApi(items: { id: number; sort_order: number }[]) {
  return requestClient.post('/admin/ext-menu/reorder', { items });
}

/** 管理员：删除扩展菜单 */
export function deleteExtMenuApi(id: number) {
  return requestClient.delete(`/admin/ext-menu/${id}`);
}

File: go-api/internal/model/ext_menu.go
package model

// ExtMenu 扩展菜单
type ExtMenu struct {
	ID        int    `json:"id"`
	Title     string `json:"title"`
	Icon      string `json:"icon"`
	URL       string `json:"url"` // 页面路径或完整 URL
	SortOrder int    `json:"sort_order"`
	Visible   int    `json:"visible"` // 1=显示 0=隐藏
	Scope     string `json:"scope"`   // frontend / backend
	CreatedAt string `json:"created_at"`
}

// ExtMenuSaveRequest 保存扩展菜单
type ExtMenuSaveRequest struct {
	ID        int    `json:"id"`
	Title     string `json:"title" binding:"required"`
	Icon      string `json:"icon"`
	URL       string `json:"url" binding:"required"`
	SortOrder int    `json:"sort_order"`
	Visible   int    `json:"visible"`
	Scope     string `json:"scope"`
}
